packages/api/portfolioDrafts/portfolio/createPortfolioStep.ts

```diff
@@ -38,12 +27,8 @@ export async function baseHandler(
   event: ApiGatewayEventParsed<PortfolioStep>,
   context?: Context
 ): Promise<APIGatewayProxyResult> {
-  // Perform shape validation
-  const setupResult = validateRequestShape<PortfolioStep>(event);
-  if (setupResult instanceof SetupError) {
-    return setupResult.errorResponse;
-  }
-  const portfolioDraftId = setupResult.path.portfolioDraftId;
+  validateRequestShape<PortfolioStep>(event);
+  const portfolioDraftId = event.pathParameters?.portfolioDraftId as string;
```

Comment:
Would this be better as
```suggestion
  const portfolioDraftId = event.pathParameters!.portfolioDraftId;
```

Or something with a `!`? Instead of the cast to a string?

Reply:
Looks like there are a few places where we alternate. Please be consistent with our behavior elsewhere in the app (whichever that is)

Reply:
> Looks like there are a few places where we alternate. Please be consistent with our behavior elsewhere in the app (whichever that is)

Don't worry, I went through all the functions that have been refactored with middy, kept them consistent.

> I sort of prefer the `!` because it forces us to add an `eslint-disable-next-line` which also encourages an explanation for _why_ we're disabling that check on that line

I disagree, the warning you would get with `const portfolioDraftId = event.pathParameters!.portfolioDraftId;` is
```
Forbidden non-null assertion.eslint@typescript-eslint/no-non-null-assertion
```
Which exists because: `event.pathParameters.portfolioDraftId;` gives a TS error 
```
Object is possibly 'null'.ts(2531)
```
It cannot be null at all because API Gateway checks if it is a required field, and `validateRequestShape<PortfolioStep>(event);` checks to see if the UUID exists and is a valid UUIDv4, which throws an error.

It was changed to `const portfolioDraftId = event.pathParameters?.portfolioDraftId as string;` because validateRequestShape no longer returns at SetupResult object with a path.

[Optional chaining TS docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining)

Reply:
Right. But I still want a comment explaining _why_ it can't be null somewhere. And if we have to do this a lot, perhaps it's an indication we need a have more abstractions around it. But I don't want us to get into the habit of ignoring nullability. The `!` forces a comment anyway. Maybe there's a better way we could expose these values into the functions using Middy so that it can definitely be type safe and so we can safely log an error if for some reason it wasn't present?

Reply:
But I guess the point I am trying to make is that I don't want to mask type errors through casting. I want to accept them and document why it's safe in this scenario.

Reply:
We could put that comment in validateRequestShape? I don't want to put it in the handler because it's so _clean_ right now.

Reply:
Pushed a commit with comment.